src/FileIO.h

```diff
@@ -60,4 +61,26 @@ struct ScopedFileDescriptor {
       close(fd);
    }
 };
+
+// Do file access as root user for the function passed in.
+// Example: auto result = FileIO::SudoFile(FileIO::ReadAsciiFileContent, filePath);
```

Comment:
"sudo" or superuser do gives the user full superuser access. Theoretically we are only elevate file system access to have root privileges, hence SudoFile. That said, I'm not opposed to changing the name. Let me know what you prefer.